### PR TITLE
Integrate styled components and fix mobile view in Bot Builder and skill wizard

### LINKED ISSUE #2399: Integrate styled components in Bot Builder
**Actual Behaviour**
Not present.

**Expected Behaviour**
Integrate styled components.

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by harsh-jindal**
Fixes #2399 

Changes: 
- [x] Integrated styled components in Bot Builder
- [x] Fix mobile view in BotBuilder
- [x] Integrated styled components in Skill Wizard
- [x] Fix mobile view in Skill Wizard

Demo Link: https://pr-2409-fossasia-susi-web-chat.surge.sh

### CODE REVIEW

**Comment by harsh-jindal** [comment]
Mobile View: Refresh the page to view the correct changes.
Screenshot 2019-07-01 at 6 48 29 PM

**Comment by harsh-jindal** [comment]
@fragm3 Resolved.

File: src/components/cms/BotBuilder/BotBuilder.js [hunk 1]
```diff
@@ -245,11 +336,8 @@ class BotBuilder extends React.Component {
         draftsOfBots.push(
           <Card
             key={draft}
-            className="bot-template-card"
             style={{
               backgroundImage: 'url(' + imageUrl + ')',
-              backgroundSize: 'cover',
-              backgroundColor: '#000',
               opacity: '0.9',
```
Comment (samagragupta) [inline src/components/cms/BotBuilder/BotBuilder.js]: You can also use `opacity: 0.9;` in styled components

**Comment by harsh-jindal** [comment]
@fragm3 @samagragupta Done.

File: src/components/cms/BotBuilder/BotWizard.js [hunk 2]
```diff
@@ -15,52 +15,159 @@ import { bindActionCreators } from 'redux';
...
+
+const ChevronRight = styled(_ChevronRight)`
+  position: absolute;
+  left: 0;
+  top: 0;
+  width: 35px;
+  height: 35px;
+  color: rgb(158, 158, 158);
+  cursor: pointer;
+  display: ${window.innerWidth} < 769 ? none : inherit;
```
Comment (fragm3) [inline src/components/cms/BotBuilder/BotWizard.js]: Use media queries

File: src/components/cms/BotBuilder/BotWizard.js [hunk 3]
```diff
@@ -15,52 +15,159 @@ import { bindActionCreators } from 'redux';
...
+
+const ChevronLeft = styled(_ChevronLeft)`
+position: absolute;
+left: 4px;
+top: 4px;
+width: 35px;
+height: 35px;
+color: white;
+cursor: pointer;
+display: ${window.innerWidth} < 769 ? none : inherit;
```
Comment (fragm3) [inline src/components/cms/BotBuilder/BotWizard.js]: Use media queries

**Comment by harsh-jindal** [comment]
@fragm3 @mariobehling Made the changes. Please review.
Now if you reload the page in mobile view, it looks like this:
Screenshot 2019-07-03 at 10 51 40 PM

**Comment by harsh-jindal** [comment]
@mariobehling The map will not be visible on surge link as `The current URL loading the Maps JavaScript API has not been added to the list of allowed referrers.`
https://developers.google.com/maps/documentation/javascript/error-messages#referer-not-allowed-map-error

Map loads perfectly on susi.ai
Screenshot 2019-07-03 at 11 19 32 PM

File: src/components/cms/BotBuilder/BotBuilder.js [hunk 4]
```diff
@@ -21,30 +20,125 @@ import {
...
+  padding: 15px 0px;
+
+  @media (min-width: 769px) {
+    padding: 15px;
+  }
+`;
+
+const CardContent = styled(_CardContent)`
+  color: #ffffff;
+  font-family: Helvetica;
```
Comment (akshatnitd) [inline src/components/cms/BotBuilder/BotBuilder.js]: Not required.

File: src/components/cms/BotBuilder/BotBuilder.js [hunk 5]
```diff
@@ -21,30 +20,125 @@ import {
...
+  font-size: 16px;
+  padding-top: 20px;
+`;
+
+const H1 = styled.h1`
+  ${commonHeadingStyle}
+`;
+
+const H2 = styled.h2`
+  ${commonHeadingStyle}
```
Comment (akshatnitd) [inline src/components/cms/BotBuilder/BotBuilder.js]: If both H1 and H2 are same, then why have you created 2 variables?

File: src/components/cms/BotBuilder/BotBuilder.js [hunk 6]
```diff
@@ -21,30 +20,125 @@ import {
...
+
+const Card = styled(_Card)`
+  width: 260px;
+  height: 170px;
+  margin: 10px;
+  overflow: hidden;
+  justify-content: center;
+  text-align: center;
+  flex-direction: row;
+  display: inline-block;
```
Comment (akshatnitd) [inline src/components/cms/BotBuilder/BotBuilder.js]: Why applying flex properties in inline-block?

File: src/components/cms/BotBuilder/BotBuilder.js [hunk 7]
```diff
@@ -21,30 +20,125 @@ import {
...
+  background-size: cover;
+  cursor: pointer;
+  background: #f2f2f2;
```
Comment (akshatnitd) [inline src/components/cms/BotBuilder/BotBuilder.js]: Why 2 background properties?

Comment (akshatnitd) [reply src/components/cms/BotBuilder/BotBuilder.js]: Infact, you have applied background properties thrice.

File: src/components/cms/BotBuilder/BotBuilderPages/DesignViews/CodeView.js [hunk 8]
```diff
@@ -18,11 +19,30 @@ import 'brace/theme/solarized_light';
 import 'brace/theme/terminal';
 import 'brace/ext/searchbox';
 
+const Container = styled.div`
+  margin-bottom: 40px;
+`;
+
+const AceEditor = styled(_AceEditor)`
```
Comment (akshatnitd) [inline src/components/cms/BotBuilder/BotBuilderPages/DesignViews/CodeView.js]: We should make a shared component for Editor. Please open an issue for it.

File: src/components/cms/BotBuilder/BotBuilderPages/DesignViews/UIView.js [hunk 9]
```diff
@@ -1,24 +1,152 @@
...
+const DesignContainer = styled.div`
+  max-width: 100%;
+  text-align: left;
+`;
+
+const FileUploadButton = styled.label`
+  border: 10px;
+  box-sizing: border-box;
+  display: inline-block;
+  font-family: Roboto, sans-serif;
```
Comment (akshatnitd) [inline src/components/cms/BotBuilder/BotBuilderPages/DesignViews/UIView.js]: Redundant and incorrect styles used. Recheck it.

File: src/components/cms/BotBuilder/BotBuilderPages/DesignViews/CodeView.js (hunk 8 above)
Comment (harsh-jindal) [reply src/components/cms/BotBuilder/BotBuilderPages/DesignViews/CodeView.js]: #2433

File: src/components/cms/BotBuilder/BotBuilder.js (hunk 5 above)
Comment (harsh-jindal) [reply src/components/cms/BotBuilder/BotBuilder.js]: One is a `h1` tag and other one is `h2` tag

**Comment by harsh-jindal** [comment]
@akshatnitd Made the changes, please review.
@mariobehling All the points stated by you above are part of SkillWizard.js and are been fixed in #2405 . So to avoid conflicts I have not made any major changes in that SkillWizard.js
Also as you mentioned about the language not working, it has been implemented in a way that first you need to select the category and then the language, and then the bot name.

**Comment by mariobehling** [comment]
Please always provide a screenshot. 

> Also as you mentioned about the language not working, it has been implemented in a way that first you need to select the category and then the language, and then the bot name.

This is a bug.

**Comment by harsh-jindal** [comment]
This bug will be fixed in this PR itself.
Screenshot:
Screenshot 2019-07-04 at 1 46 26 PM

**Comment by harsh-jindal** [comment]
The above issue will be fixed in the following PR. #2441

**Comment by harsh-jindal** [comment]
@mariobehling @akshatnitd @fragm3 Please review.
The spacing issue at the bottom in skill wizard is due to having a `padding-bottom: 120px` in App.js . Also the overlapping footer in botbuilder in mobile view is due to the same reason. I have added a work around so that the footer does not overlap in botbuilder.
The footer issue is generic and shall be addressed in a separate PR.

**Comment by harsh-jindal** [comment]
@samagragupta Can you provide a screenshot?

**Comment by harsh-jindal** [comment]
@samagragupta Fixed

File: src/components/cms/BotBuilder/BotBuilder.js [hunk 10]
```diff
@@ -21,30 +20,117 @@ import {
...
+`;
+
+const BotContainer = styled.div`
+  display: flex;
+  flex-wrap: wrap;
+  flex-direction: row;
+  margin: 10px;
+
+  @media (max-width: 580px) {
+    display: flex;
```
Comment (fragm3) [inline src/components/cms/BotBuilder/BotBuilder.js]: Remove display: flex

File: src/components/cms/BotBuilder/BotBuilderPages/Configure.js [hunk 11]
```diff
@@ -9,14 +9,43 @@ import CodeView from './ConfigureViews/CodeView';
...
 import './Animation.min.css';
+import styled from 'styled-components';
+
+const Home = styled.div`
+  margin-top: 10px;
+`;
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: row;
```
Comment (fragm3) [inline src/components/cms/BotBuilder/BotBuilderPages/Configure.js]: flex-direction default value is `row`. Remove it.

File: src/components/cms/BotBuilder/BotBuilderPages/Design.js [hunk 12]
```diff
@@ -9,15 +9,47 @@ import CodeView from './DesignViews/CodeView';
...
+import styled from 'styled-components';
+
+const Home = styled.div`
+  text-align: center;
+  margin-top: 10px;
+`;
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: row;
```
Comment (fragm3) [inline src/components/cms/BotBuilder/BotBuilderPages/Design.js]: Remove

**Comment by harsh-jindal** [comment]
@fragm3 As mentioned by @akshatnitd in the PR #2405 https://github.com/fossasia/susi.ai/pull/2405#pullrequestreview-256832072
The save and cancel button shall be center aligned.
Fixing the other things.

**Comment by harsh-jindal** [comment]
@fragm3 Fixed. Please review

**Comment by akshatnitd** [comment]
@harsh-jindal Make it right-aligned. I didn't notice that in my previous review.

**Comment by harsh-jindal** [comment]
@akshatnitd @fragm3 Fixed.

**Comment by mariobehling** [comment]
Looks good to me. We can also follow up on more points in other issues. Thanks.

**Pull request merged**